platforms/scala/index.js

```diff
@@ -18,3 +26,17 @@ module.exports = {
     return compiled({});
   }
 };
+
+function crossPlatformDir(dir) {
+  if (process.platform == 'win32') {
+    // this is due to how volumes are mounted by boot2docker/virtualbox
+    var p = path.parse(dir);
+    return path.posix.sep + p.root.split(':')[0].toLowerCase() + path.posix.sep
+    + p.dir.substring(p.root.length).replaceAll(path.sep, path.posix.sep) + path.posix.sep + p.base;
+  }
+  return dir;
+}
+
+function crossPlatformUserHome() {
+  return process.env[(process.platform == 'win32') ? 'USERPROFILE' : 'HOME'];
+}
```

Comment:
these funcs need to be commonly accessible to all platforms. is there a way to do that?

I also need common funcs for the java, clojure, etc platforms.